File: results/py/HOW_TO_CHECK_IF_A_GIVEN_ARRAY_REPRESENTS_A_BINARY_HEAP.cpp
bool f_filled ( int * arr , int i , int n ) {
  if ( i > ( int ) ( ( n - 2 ) / 2 ) ) return true ;
  if ( ( arr [ i ] >= arr [ 2 * i + 1 ] && arr [ i ] >= arr [ 2 * i + 2 ] && f_filled ( arr , 2 * i + 1 , n ) && f_filled ( arr , 2 * i + 2 , n ) ) return true ;
  return false ;
}

File: TransCoder_tokenized_test_set_functions/python/HOW_TO_CHECK_IF_A_GIVEN_ARRAY_REPRESENTS_A_BINARY_HEAP.py
def isHeap ( arr, i, n ) :
    if i > int ( ( n - 2 ) / 2 ) :
        return True
    if ( arr [ i ] >= arr [ 2 * i + 1 ] and arr [ i ] >= arr [ 2 * i + 2 ] and isHeap ( arr, 2 * i + 1, n ) and isHeap ( arr, 2 * i + 2, n ) ) :
        return True
    return False
